hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/replicas/ReplicaKey.java

```diff
@@ -40,9 +40,10 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (!(o instanceof ReplicaKey other)) {
```

Comment:
This change adheres to our restrictions around new language features, please see https://github.com/apache/hbase/pull/6729